spec/docs/expressions.md

```diff
@@ -4,31 +4,37 @@
 
 ## Expression map (`rml:ExpressionMap`)
 
-An <dfn>expression map</dfn> (`rml:ExpressionMap`) is an abstract class, that is specialized by other RML classes. An [=expression map=] MAY have one the following properties:
-* 0 or 1 `rml:constant`, or
-* 0 or 1 `rml:reference`, or
-* 0 or 1 `rml:template`.
+An <dfn>expression map</dfn> (`rml:ExpressionMap`) is an abstract class, that is specialized by other RML classes. An [=expression map=] SHOULD have one the following properties:
+* 1 `rml:constant`, or
+* 1 `rml:reference`, or
+* 1 `rml:template`.
 
-Each of these properties specifies an [=expression=] which, upon evaluation, results in a set of values.
+Each of these properties specifies an [=expression=] which, upon evaluation, results in an ordered list of values.
 
-The <dfn>reference expression set</dfn> of an [=expression map=] ar the set of expressions which are evaluated on a [=logical iteration=].
+The <dfn>set of values</dfn> are the values of the [=references expression set=] which are returned by an [=expression=].
+
+For each value in the ordered list of values, an [=expression value=] is created. If the ordered list of values is empty, `NULL` is returned and no [=expression value=] is created.
+
+The <dfn>reference expression set</dfn> of an [=expression map=] is the set of expressions which are evaluated on a [=logical iteration=].
```

Comment:
This does not belong here. What happens with an "expression value" depends on how the expression map is used.